gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java

```diff
@@ -132,6 +143,50 @@ private static void updateBasedOnReferenceToReferenceDuplicatedPlan(List<Referen
         }
     }
 
+    private @NotNull MedicationStatement duplicateMedicationStatementForOrderBasedOnSharedAcutePlan(
+        List<MedicationStatement> medicationStatements,
+        MedicationRequest order,
+        MedicationRequest originalPlan,
+        MedicationRequest duplicatedPlan
+    ) {
+
+        var originalMedicationStatement = medicationStatements
+            .stream()
+            .filter(medicationStatement -> medicationStatement.getId().equals(originalPlan.getId() + "-MS"))
+            .findFirst()
+            .orElseThrow(() ->
+                // there should always be a medication statement associated with an order, but we have to handle
+                // the possibility it is not found.
+                new IllegalStateException(
```

Comment:
In Java it's usually best practice to create a `RuntimeException` which suits your specific need. `IllegalStateException` suggests that [a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation](https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalStateException.html#:~:text=Signals%20that%20a%20method%20has%20been%20invoked%20at%20an%20illegal%20or%20inappropriate%20time.%20In%20other%20words%2C%20the%20Java%20environment%20or%20Java%20application%20is%20not%20in%20an%20appropriate%20state%20for%20the%20requested%20operation.).